### ISSUE TITLE
Native crash of libandroidbridge.so on Android high version system

### ISSUE THREAD

**Issue by SwayChen** [strongswan/strongswan#1504]
❗️This is a native crash that I can't reproduce on my device. It was tracked by Google Play console vitals

**strongswan version**: 5.7.2
**Crash Android version**: high version Android device (Android 12 especially)

<img width="687" alt="image" src="https://user-images.githubusercontent.com/6345749/211486553-80b00353-c5ba-479e-88e3-e8a462081aa5.png">

**Crash backtrace**

<img width="779" alt="image" src="https://user-images.githubusercontent.com/6345749/211488593-33d12803-6e62-4840-8a48-18b4736f8f36.png">

Tread: `FORTIFY: FD_SET: file descriptor 3421 >= FD_SETSIZE 1024`
```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 12832 >>> {PACKAGE_NAME} <<<

backtrace:
  #00  pc 0x0000000000078b9c  /apex/com.android.runtime/lib64/bionic/libc.so (abort+180)
  #01  pc 0x000000000007a9fc  /apex/com.android.runtime/lib64/bionic/libc.so (__fortify_fatal(char const*, ...)+124)
  #02  pc 0x00000000000a23b8  /apex/com.android.runtime/lib64/bionic/libc.so (__FD_SET_chk+120)
  #03  pc 0x0000000000008770  /data/app/~~XvR4eVYN9X9wrpso7N9iVw==/{PACKAGE_NAME}-BKv48nJf11dhzjDpqCyRCQ==/lib/arm64/libandroidbridge.so
  #04  pc 0x000000000007cef8  /data/app/~~XvR4eVYN9X9wrpso7N9iVw==/{PACKAGE_NAME}-BKv48nJf11dhzjDpqCyRCQ==/lib/arm64/libstrongswan.so
  #05  pc 0x000000000008f708  /data/app/~~XvR4eVYN9X9wrpso7N9iVw==/{PACKAGE_NAME}-BKv48nJf11dhzjDpqCyRCQ==/lib/arm64/libstrongswan.so
  #06  pc 0x00000000000dc188  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+264)
  #07  pc 0x000000000007a438  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
```

**crash dump by `ndk-stack`**
```
********** Crash dump: **********
#00 0x0000000000078b9c /apex/com.android.runtime/lib64/bionic/libc.so (abort+180)
#01 0x000000000007a9fc /apex/com.android.runtime/lib64/bionic/libc.so (__fortify_fatal(char const*, ...)+124)
#02 0x00000000000a23b8 /apex/com.android.runtime/lib64/bionic/libc.so (__FD_SET_chk+120)
#03 0x0000000000008770 /data/app/~~XvR4eVYN9X9wrpso7N9iVw==/{PACKAGE_NAME}-BKv48nJf11dhzjDpqCyRCQ==/lib/arm64/libandroidbridge.so
handle_plain
/strongswan/src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c:213:2
#04 0x000000000007cef8 /data/app/~~XvR4eVYN9X9wrpso7N9iVw==/{PACKAGE_NAME}-BKv48nJf11dhzjDpqCyRCQ==/lib/arm64/libstrongswan.so
process_job
/strongswan/src/frontends/android/app/src/main/jni/../../../../../../../src/libstrongswan/processing/processor.c:235:13
process_jobs
/strongswan/src/frontends/android/app/src/main/jni/../../../../../../../src/libstrongswan/processing/processor.c:321:0
#05 0x000000000008f708 /data/app/~~XvR4eVYN9X9wrpso7N9iVw==/{PACKAGE_NAME}-BKv48nJf11dhzjDpqCyRCQ==/lib/arm64/libstrongswan.so
thread_main
/strongswan/src/frontends/android/app/src/main/jni/../../../../../../../src/libstrongswan/threading/thread.c:331:8
#06 0x00000000000dc188 /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+264)
#07 0x000000000007a438 /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
```
According the dump stack, it points to `FD_SET(tunfd, &set);`, I can't explore any solution

**Comment by tobiasbrunner** [comment]
> According the dump stack, it points to `FD_SET(tunfd, &set);`, I can't explore any solution

If you'd searched for that, you'd have found #1160 with the fix in 7433f1672a49325eb4e8f2362248828911173da0.

**Comment by SwayChen** [comment]
Thanks, I found it already using another keyword...

**Comment by zoobibackups** [comment]
> Thanks, I found it already using another keyword...

caan you show your solution here

**Comment by zoobibackups** [comment]
> > According the dump stack, it points to `FD_SET(tunfd, &set);`, I can't explore any solution
> 
> If you'd searched for that, you'd have found #1160 with the fix in [7433f16](https://github.com/strongswan/strongswan/commit/7433f1672a49325eb4e8f2362248828911173da0).

we are facing the same issue. can you paste your solution here

**Comment by zoobibackups** [comment]
> > According the dump stack, it points to `FD_SET(tunfd, &set);`, I can't explore any solution
> 
> If you'd searched for that, you'd have found #1160 with the fix in [7433f16](https://github.com/strongswan/strongswan/commit/7433f1672a49325eb4e8f2362248828911173da0).

this does not worked. as excepted. still we are facing a crash with multiple reconnects and connects.

**Comment by AftabUfaq** [comment]
after adding this pull request #2404 this issue was fixed
